### PR TITLE
Fix path conflicts in chip_data_model in some projects

### PR DESCRIPTION
**Pull request by mspang**
### Problem

The chip_data_model template builds a library that contains 2 files
named encoder.cpp and two files name attribute-size.cpp.

While this is not a problem with pigweed's toolchain definitions because
different directories are used, it can cause a path conflict if a
project defines a toolchain that places all of a library's .o files in
the same directory:

```
ERROR at //third_party/connectedhomeip/repo/src/app/chip_data_model.gni:59:3: Duplicate object file
  source_set(_data_model_name) {
  ^-----------------------------
The target //third_party/connectedhomeip/repo/src/controller/data_model:data_model
generates two object files with the same name:
  obj/third_party/connectedhomeip/repo/src/controller/data_model/data_model/attribute-size.o
```

### Change overview

Rename some files so that all files in the same library have distinct
filenames, which should be compatible with most reasonable toolchain
definitions.

### Testing

Generate the project with a toolchain that places all of a library's .o files in the same directory.

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

**Review by woody-apple — CHANGES_REQUESTED** [review]
Can we update this to conform to the standard template?

**Review by woody-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

File: src/darwin/Framework/CHIP.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -98,7 +98,7 @@
 		1E85731C26551A490050A4D9 /* DataModelHandler.cpp */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.cpp; name = DataModelHandler.cpp; path = ../../../app/util/DataModelHandler.cpp; sourceTree = "<group>"; };
 		1E85731D26551A490050A4D9 /* util.cpp */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.cpp; name = util.cpp; path = ../../../app/util/util.cpp; sourceTree = "<group>"; };
 		1E85731E26551A490050A4D9 /* ember-print.cpp */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.cpp; name = "ember-print.cpp"; path = "../../../app/util/ember-print.cpp"; sourceTree = "<group>"; };
-		1E85731F26551A490050A4D9 /* attribute-size.cpp */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.cpp; name = "attribute-size.cpp"; path = "../../../app/util/attribute-size.cpp"; sourceTree = "<group>"; };
+		1E85731F26551A490050A4D9 /* attribute-size.cpp */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.cpp; name = "attribute-size-util.cpp"; path = "../../../app/util/attribute-size-util.cpp"; sourceTree = "<group>"; };
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP.xcodeproj/project.pbxproj:101]:
```suggestion
		1E85731F26551A490050A4D9 /* attribute-size-util.cpp */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.cpp; name = "attribute-size-util.cpp"; path = "../../../app/util/attribute-size-util.cpp"; sourceTree = "<group>"; };
```

**Pull request merged**